### PR TITLE
faucet: remove race condition when getting sequence numbers

### PR DESCRIPTION
**Pull request by bmwill**
The current implementation of Faucet is racy, in that 2 independent requests can come in
and result in faucet sending 2 txns to the network with the same sequence number and
they race to complete. One request will complete while the other will get a HashMismatch
Error.

This PR fixes that by ensuring that sequence numbers are not reused.

### CODE REVIEW

File: client/faucet/src/mint.rs [hunk 1]
```diff
@@ -109,83 +64,107 @@ impl MintParams {
...
+                ),
+            ));
+            txns
+        };
 
         let batch = txns
             .iter()
             .map(|txn| MethodRequest::submit(txn))
             .collect::<Result<_, _>>()?;
         self.client.batch(batch).await?;
```
Comment (xli) [inline client/faucet/src/mint.rs:156]: whenever there is error returned, or any other problem causes transaction is not executed, we may have a sequence number not used, which will block all the following transactions, as account sequence number is never rollback to an old number, faucet will stop work.

File: client/faucet/src/main.rs [hunk 2]
```diff
@@ -204,7 +208,6 @@ mod tests {
                 )
                 .reply(&filter)
                 .await;
-            assert_eq!(resp.body(), "1"); // 0+1
```
Comment (xli) [inline client/faucet/src/main.rs:207]: something changed? we want to ensure the response number is expected, right?

Comment (bmwill) [reply client/faucet/src/main.rs:207]: Yeah i'll fix this

File: client/faucet/src/mint.rs (hunk 1 above)
Comment (bmwill) [reply client/faucet/src/mint.rs]: Good point, generally this probably shouldn't happen, but i'll add some logic to reset the local sequence number upon error

File: client/faucet/src/mint.rs [hunk 3]
```diff
@@ -109,83 +64,114 @@ impl MintParams {
...
+                    vec![],
+                ),
+            ));
+            txns
+        };
 
         let batch = txns
             .iter()
             .map(|txn| MethodRequest::submit(txn))
             .collect::<Result<_, _>>()?;
```
Comment (xli) [inline client/faucet/src/mint.rs:155]: Nit, maybe unwrap is better as we don't expect an error here and panic will be better option if it happened

**Review by xli — APPROVED** [review]

**Pull request closed without merging**